packages/core/src/badge/Badge.tsx

```diff
@@ -42,6 +42,7 @@ export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(function Badge(
       <span
         className={clsx(withBaseName("badge"), {
           [withBaseName("topRight")]: children,
+          [withBaseName("dotBadge")]: !value,
```

Comment:
as we are determining the type of badge through value, then if someone provided 0, it would switch to a dotless badge

Reply:
fixed